opm/parser/eclipse/RawDeck/StarToken.hpp

```diff
@@ -36,15 +36,39 @@ namespace Opm {
             return boost::lexical_cast<T>(valueString);
         }
         catch (boost::bad_lexical_cast&) {
-            throw std::invalid_argument("Unable to parse string" + valueString + " to typeid: " + typeid(T).name());
+            throw std::invalid_argument("Unable to convert string '" + valueString + "' to typeid: " + typeid(T).name());
         }
     }
 
     template <>
+    inline double readValueToken<double>(const std::string& valueString) {
+        // Eclipse supports Fortran syntax for specifying exponents of floating point
+        // numbers ('D' and 'E', e.g., 1.234d5) while C++ only supports the 'e' (e.g.,
+        // 1.234e5). the quick fix is to replace 'D' by 'E' and 'd' by 'e' before trying
+        // to convert the string into a floating point value. This may not be the most
+        // performant thing to do, but it is probably fast enough.
+        std::string vs(valueString);
+        std::replace(vs.begin(), vs.end(), 'D', 'E');
+        std::replace(vs.begin(), vs.end(), 'd', 'e');
+
+        try {
+            return boost::lexical_cast<double>(vs);
```

Comment:
I haven't measured, but parsing the Norne deck in debug mode takes about 5 seconds on my machine. I suppose that less than 5% of that time are spend in `boost::lexical_cast`, so I think `strtod` would only save a few milliseconds. other optimizations -- like reducing the use of `std::shared_ptr` -- would probably provide a drastically better return of investment. (This is only my gut-feeling, though...)